### PR TITLE
fix: add rate limit on telegram api and split messages by unicode

### PR DESCRIPTION
**Pull request by zenixls2**
- add rate limit in telegram.
- split messages by size limit and unicode header.

not strictly following the size limit (4096 bytes) and rate limit (100req/s) written in telegram api doc.  
use 3000, 10 instead.

### CODE REVIEW

**Review by austin362667 — APPROVED** [review]

File: pkg/interact/telegram.go [hunk 1]
```diff
@@ -62,7 +68,14 @@ type TelegramReply struct {
 }
 
 func (r *TelegramReply) Send(message string) {
-	checkSendErr(r.bot.Send(r.session.Chat, message))
+	var left, right int
+	for left, right = 0, maxMessageSize; right < len(message); left, right = right, right+maxMessageSize {
```
Comment (c9s) [inline pkg/interact/telegram.go:72]: you only need to split the message when its length > maxMessageSize?

File: pkg/interact/telegram.go [hunk 2]
```diff
@@ -158,7 +171,24 @@ func (tm *Telegram) Start(context.Context) {
 		if reply.set {
 			reply.build()
 			if len(reply.message) > 0 || reply.menu != nil {
-				checkSendErr(tm.Bot.Send(m.Chat, reply.message, reply.menu))
+				var left, right int
+				for left, right = 0, maxMessageSize; right < len(reply.message); left, right = right, right+maxMessageSize {
```
Comment (c9s) [inline pkg/interact/telegram.go:175]: for the split thing, can you pull out it to a function?

File: pkg/interact/telegram.go (hunk 1 above)
Comment (zenixls2) [reply pkg/interact/telegram.go]: yes

File: pkg/interact/telegram.go (hunk 2 above)
Comment (zenixls2) [reply pkg/interact/telegram.go]: done

**Review by c9s — APPROVED** [review]

**Pull request merged**